otherlibs/systhreads/st_stubs.c

```diff
@@ -729,10 +732,9 @@ CAMLprim value caml_thread_yield(value unit)
   */
 
   caml_raise_if_exception(caml_process_pending_signals_exn());
-  caml_thread_save_runtime_state();
+  save_runtime_state();
   st_thread_yield(m);
-  Active_thread = st_tls_get(caml_thread_key);
```

Comment:
Is it really ok to remove this line? Once the thread is back in control, `Active_thread` may not be `This_thread`

Reply:
This is simply moving the read of the TLS entry to the line introduced just below it (which sets the right invariant for Active_thread in a single location).